### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by jpc20**
Fix error where we dont pass a instance variable to the new user form and it errors out looking for one.
Also got factory bot working.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -5,42 +5,30 @@ def show
   end
 
   def new
-
+    return @user if params[:user]
+    @user = Hash.new("")
```
Comment (Lithnotep) [inline app/controllers/users_controller.rb:9]: Awesome fix

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -5,42 +5,30 @@ def show
...
       flash[:message] = "Hi #{user.name} you have registered and logged in!"
       session[:user_id] = user.id
       redirect_to "/profile"
-    elsif !empty_params.empty?
-      flash[:notice] = "Please fill in #{empty_params.keys.join(", ")}"
-      redirect_to '/register'
+    elsif user.errors.added?(:email, :taken)
+      flash[:message] = "This email already exists"
+      @user = user_params
+      render :new
```
Comment (Lithnotep) [inline app/controllers/users_controller.rb:21]: Much cleaner great change

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -5,42 +5,30 @@ def show
...
   end
 
   def create
     user = User.new(user_params)
-    empty_params = user_params.select{|param, value| value.empty?}
     if user.save
       flash[:message] = "Hi #{user.name} you have registered and logged in!"
       session[:user_id] = user.id
       redirect_to "/profile"
-    elsif !empty_params.empty?
```
Comment (janegreene) [inline app/controllers/users_controller.rb:18]: replaced with active record solution

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -25,7 +25,7 @@
 # directory. Alternatively, in the individual `*_spec.rb` files, manually
 # require only the support files necessary.
 #
-# Dir[Rails.root.join('spec', 'support', '**', '*.rb')].each { |f| require f }
```
Comment (janegreene) [inline spec/rails_helper.rb:28]: Great catch, glad this is working.

**Pull request merged**